packages/styling/src/styles/getFadedOverflowStyle.ts

```diff
@@ -0,0 +1,113 @@
+import { IRawStyle } from '@uifabric/merge-styles';
+import { HighContrastSelector } from './CommonStyles';
+import { ITheme } from '../interfaces';
+
+export interface IRGB {
+  r: number;
+  g: number;
+  b: number;
+}
+
+const DEFAULT_HEIGHT = '50%';
+const DEFAULT_WIDTH = 20;
+
+/**
+ * - Generates a style used to fade out an overflowing content by defining a style for an :after pseudo element.
+ * - Apply it to the :after selector for all combination of states the parent of content might have (normal, hover, selected, focus).
+ * - Requires the target to have position set to relative and overflow set to hidden.
+ *
+ * @example
+ * ```tsx
+ * // Assuming the following DOM structure and the different background colors coming from the parent holding the content.
+ * <div className={classNames.parent}>
+ *   <span className={classNames.content}>Overflown Content</span>
+ * </div>
+ * ```
+ * ```ts
+ * // This is how the style set would look in Component.styles.ts
+ * const {bodyBackground, hoverBackground} = theme.semanticColors;
+ *
+ * const styles = {
+ *   parent: [
+ *     backgroundColor: bodyBackground,
+ *     selectors: {
+ *       '&:hover: {
+ *         backgroundColor: hoverBackground
+ *       },
+ *       '$content:after': {
+ *         ...getFadedOverflowStyle(theme, bodyBackground)
+ *       },
+ *       '&:hover $content:after': {
+ *         ...getFadedOverflowStyle(theme, hoverBackground)
+ *       }
+ *     }
+ *   ],
+ *   content: [
+ *     width: '100%',
+ *     display: 'inline-block',
+ *     position: 'relative',
+ *     overflow: 'hidden'
+ *   ]
+ * }
+ * ```
+ * @param theme - The theme object to use.
+ * @param color - The background color to fade out to. Defaults to theme.semanticColors.bodyBackground.
+ * @param direction - The direction of the overflow. Defaults to horizontal.
+ * @param width - The width of the fading overflow. Vertical direction defaults it to 100% vs 20px when horizontal.
+ * @param height - The Height of the fading overflow. Vertical direction defaults it to 50% vs 100% when horizontal.
+ * @returns The style object.
+ */
+export function getFadedOverflowStyle(
+  theme: ITheme,
+  color: string = theme.semanticColors.bodyBackground,
```

Comment:
this should be something in typeof ISemanticColors (whatever the syntax is for that)

Reply:
@phkuo user is passing only the value of a semantic color which is a string... don't understand what do you mean by typeof ISemanticColors. Am I missing something?

Reply:
@Vitalius1 we have other places (see the text thing in experiments) which enforce that you pass in a semantic slot, so you're actually passing in a type. And then it will get that slot from the theme for you. It's a way to force users to use only theme colors, instead of custom ones. You can consider doing that here, instead of allowing raw colors. Up to you.